### PR TITLE
UCS/SYS: Implement ucs_get_tid() and ucs_tgkill() using FreeBSD tids.

### PR DESCRIPTION
**Pull request by kostikbel**
Signed-off-by: Konstantin Belousov <konstantinb@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by hoopoepg** [comment]
ok to test

File: src/ucs/sys/sys.c [hunk 1]
```diff
@@ -965,12 +968,27 @@ ucs_status_t ucs_sys_fcntl_modfl(int fd, int add, int remove)
...
 {
+#ifdef SYS_gettid
     return syscall(SYS_gettid);
+#elif defined(HAVE_SYS_THR_H)
+    long id;
+
+    thr_self(&id);
+    return (id);
+#else
+#error "Port me"
```
Comment (hoopoepg) [inline src/ucs/sys/sys.c]: nice message :)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/11536/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 29 workers (click for details)</summary>

**Note:** the logs will be deleted after 02-Aug-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/131d9aca8670d05316d408ccf5e01b7cdef79658/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/18d381d8f9f397d8a24cc151072634e6dd512e54/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/5aa24b2949f5d7b5b7929322609cb0b71db403f1/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/adce3a39c470d18b9c13223e98e2334272c39093/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/46881490627cc38c4797123466b7bcabb19f5411/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/cfcbce8942df01318239b8990287446653a3c141/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/e1495c23ff10f8500f3ccccb48c3324b37cedb14/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/85f8a25f3c6e37738f94b3691a1dc3326ed1e7ed/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/3ddb2d2f581a6d062cd506d3ffce7e9a31163f7b/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W0](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/f9ffd52628c3794b7e03569fd66075942c5c4eaf/hpc-test-althca_W0) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W1](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/7a699bd452330db6d171b82f67d6e5b7cd41abcd/hpc-test-althca_W1) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W2](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/0a1554e68db89d3ac14d1c43895d8c789a045e97/hpc-test-althca_W2) | :heavy_check_mark: SUCCESS
[hpc-test-althca_W3](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/fd343b075fe63bebe0b466dc3911339c033715b2/hpc-test-althca_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/511bd641dc29ac09e36c51296abeffe17172213f/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/5338e9f56ebb84930f9eb62b16c548c0d686d6ae/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/2868a88f18a325fb9e60adf8ac5c9bc280621210/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/31a7e565360d3fe35c293e55a5b4b8beb0b5916b/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/bc4a415d250945853c6c538ce4ec28645c0c74ec/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/0ff9e75e21696ea5a92a75195884d42ff2e04603/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/14616545875bd3847d8e3bfd439c69df9717616f/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/f4d6cc78056529efade55916975b69099fc8bcb7/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/bbdc99fda680cf5d7b013f9a55593cf99afca042/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/2d181a355416a7b29f5c1db49889f66404ffae95/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/edc5c99675d3a2864a621916a61fe0559ff65006/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/1efe412cbc5b91a561f2530b75326527636d72a3/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/5f6fb1827558c49ec3d5cc31100cb01dc8a25402/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/03fc80ea1cad97f416ef827f17982935ce9bad1c/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/003b4f73d2d906057ab785d14345a935bdb76a10/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/338edc914d2b32293d9387a5a543a404/raw/565aaa9f1812dcd68e493344626dd3119ee84e55/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**